src/pyinsole/routes.py

```diff
@@ -2,15 +2,20 @@
 import logging
 from typing import Callable
 
+from .translators import AbstractTranslator, TranslatedMessage
+from .providers import AbstractProvider
+from .handlers import Handler, AbstractHandler
 from .compat import iscoroutinefunction
-from .handlers import ICallable, IHandler
-from .providers import IProvider
-from .translators import ITranslatedMessage, ITranslator
 
 logger = logging.getLogger(__name__)
 
 
-async def to_coroutine(func, *args, **kwargs):
+async def to_coroutine(handler: Handler, *args, **kwargs):
+    func = handler
+
+    if isinstance(handler, AbstractHandler):
```

Comment:
Talvez algo como:

```suggestion
    if callable(handler):
```

Reply:
Estava pensando só em trocar o `isinstance(handler, AbstractHandler)` por verificar se é um callable. Mas sim, vai ter que continuar com a lógica do `__call__`

Reply:
Fiz um teste aqui, e mesmo que for uma função simples, existe o `__call__`. Então passar aqui `iscoroutinefunction(func.__call__)` já deve cobrir todos os casos.